studio/pages/new/[slug].tsx

```diff
@@ -20,40 +27,99 @@ import {
   PASSWORD_STRENGTH,
   PASSWORD_STRENGTH_COLOR,
   PASSWORD_STRENGTH_PERCENTAGE,
+  PRICING_PLANS,
+  PRICING_PLANS_DEFAULT,
 } from 'lib/constants'
 
 import { useStore, withAuth } from 'hooks'
+import { IRootStore } from 'stores'
 import { WizardLayout } from 'components/layouts'
 import FormField from 'components/to-be-cleaned/forms/FormField'
 import Panel from 'components/to-be-cleaned/Panel'
 import InformationBox from 'components/ui/InformationBox'
+import { Organization } from 'types'
+import { getURL } from 'lib/helpers'
+
+interface StripeCustomer {
+  paymentMethods: any
+  customer: any
+  error?: any
+}
 
 interface IHomePageStore {
-  store: any
-  projectOrganizations: Dictionary<any>[]
+  store: IRootStore
+  organizations: Dictionary<any>[]
+  currentOrg: Organization | undefined
+  isEmptyOrganizations: boolean
+  isEmptyPaymentMethod: boolean | undefined
+  isInvalidSlug: boolean
+  isOverFreeProjectLimit: boolean
+  stripeCustomerId: string | undefined
+  loadStripeAccount: () => void
 }
 class HomePageStore implements IHomePageStore {
-  store: any
+  store: IRootStore
+  router: NextRouter
 
-  constructor(store: any) {
+  stripeCustomerError: Error | undefined
+  stripeCustomer: StripeCustomer | undefined
+
+  constructor(store: IRootStore, router: NextRouter) {
     makeAutoObservable(this)
     this.store = store
+    this.router = router
+  }
+
+  get organizations() {
+    return values(toJS(this.store.app.organizations.list()))
+  }
+
+  get currentOrg() {
+    const { slug } = this.router.query
+    return this.organizations.find((o: any) => o.slug === slug)
+  }
+
+  get stripeCustomerId() {
+    return this.currentOrg?.stripe_customer_id
   }
 
-  get projectOrganizations() {
-    const projects = values(this.store?.projects?.data) ?? []
-    const orgIds = projects.map((x: any) => x.organization_id)
-    const temp = values(this.store?.organizations?.data) ?? []
-    return temp
-      .filter((x: any) => orgIds.includes(x.id))
-      .sort((a: any, b: any) => a.name.localeCompare(b.name))
+  get isEmptyOrganizations() {
+    return this.organizations.length <= 0
+  }
+
+  get isEmptyPaymentMethod() {
+    if (!this.stripeCustomer?.paymentMethods) return undefined
+    return this.stripeCustomer?.paymentMethods?.data?.length <= 0
+  }
+
+  get isInvalidSlug() {
+    return isUndefined(this.currentOrg)
+  }
+
+  get isOverFreeProjectLimit() {
+    const freeProjects = this.currentOrg?.total_free_projects ?? 0
```

Comment:
this has been updated elsewhere. 
I think there is a total_free_projects available in user profile.

Reply:
in https://github.com/supabase/infrastructure/pull/3130/files user.total_free_projects_owned is getting added.

Reply:
yup ^ total_free_projects will come with this PR:
https://github.com/supabase/infrastructure/pull/3130

Reply:
perhaps these PR should go in first?
https://github.com/supabase/infrastructure/pull/3130
https://github.com/supabase/supabase/pull/4586